tests/lib/nested.sh

```diff
@@ -1035,6 +1064,9 @@ nested_start_core_vm_unit() {
     # wait for the nested-vm service to appear active
     wait_for_service "$NESTED_VM"
 
+    # make sure the service started and it is running
+    nested_check_boot_errors
```

Comment:
I think it is slightly unexpected (given the name of this function - "check") that it may actually silently restart the VM - as such it can only be used in certain contexts. Maybe the function should really just check for vm errors, and restart should be up to the caller? For example:
```
if nested_check_boot_errors; then
    nested_restart
fi
if ! nested_check_unit_active "$NESTED_VM" 
...
```

Reply:
Agree, change done, thanks for the review